www/js/lib/uiUtil.js

```diff
@@ -32,6 +32,38 @@ if (webpMachine) {
 }
 
 define(rqDef, function(settingsStore) {
+
+    /**
+     * Displays a Bootstrap alert or confirm dialog box depending on the options provided
+     * 
+     * @param {String} label The modal's label or title which appears in the header 
+     * @param {String} message The alert message to display in the body of the modal 
+     * @param {Boolean} isConfirm If true, the modal will be a confirm dialog box, otherwise it will be an alert 
+     * @param {String} declineButtonText The text to display on the decline button(optional, Default = "Cancel") 
+     * @param {String} approveButtonText  The text to display on the approve button(optional, Default = "Confirm")
+     * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel, backdrop or the cross(x) button
+     */
+    function systemAlert(label, message, isConfirm, declineButtonText = "Cancel", approveButtonText = "Confirm") {
```

Comment:
Did you check whether this syntax works in IE11 (the baseline that, although we have deprecated support for, we still try to keep going if not too complicated)? If you don't have IE11 but do have Microsoft Edge, then MS Edge has an Internet Explorer Mode, which uses the IE11 engine.

I seem to remember that adding default values might not work in that browser at least, possibly in some other old ones. But I may be wrong. If not, simply have a line below with `declineButtonText = declineButtonText || 'Cancel'`, and similar for `approveButtonText`.

Reply:
I'm observing a bit of weird behavior in the IE mode. I can't even access the configure page of the extensions when I am on the `Use-bootstrap-modal-for-popups-#798` branch. The issue ain't a problem on the master branch. In fact, the extension doesn't load properly when I am on the `Use-bootstrap-modal-for-popups-#798` branch as the Kiwix logo says just 'Kiwix' instead of the 'Kiwix 3.3-WIP' as it should. What could go wrong here?

Reply:
I'll check.